content/en/blog/_posts/2025-02-03-introducing-jobset/2025-02-03-introducing-jobset.md

```diff
@@ -0,0 +1,211 @@
+not adequately model them alone anymore. 
+
+Furthermore, the landscape of Kubernetes distributed training orchestration APIs has become
+fragmented, and each of the existing solutions in this fragmented landscape has certain limitations
+that make it non-optimal for distributed ML training. 
+
+For example, the KubeFlow training operator defines custom APIs for different frameworks (e.g.
+PyTorchJob, TFJob, MPIJob, etc.); however, each of these job types are in fact a solution fit
+specifically to the target framework, each with different semantics and behavior. 
+
+On the other hand, the Job API fixed many gaps for running batch workloads, including Indexed
+completion mode, higher scalability, Pod failure policies and Pod backoff policy to mention a few of
+the most recent enhancements. However, running ML training and HPC workloads using the upstream Job
+API requires extra orchestration to fill the following gaps:
+
+Multi-template Pods : Most HPC or ML training jobs include more than one type of Pods. The different
+Pods are part of the same workload, but they need to run a different container, request different
+resources or have different failure policies. A common example is the driver-worker pattern.
+
+Job groups : Large scale training workloads span multiple network topologies, running across
+multiple racks for example. Such workloads are network latency sensitive, and aim to localize
+communication and minimize traffic crossing the higher-latency network links. To facilitate this,
+the workload needs to be split into groups of Pods each assigned to a network topology.
+
+Inter-Pod communication : Create and manage the resources (e.g. [headless
+Services](/docs/concepts/services-networking/service/#headless-services)) necessary to establish
+communication between the Pods of a job.
+
+Startup sequencing : Some jobs require a specific start sequence of pods; sometimes the driver is
+expected to start first (like Ray or Spark), in other cases the workers are expected to be ready
+before starting the driver (like MPI).
+
+JobSet aims to address those gaps using the Job API as a building block to build a richer API for
+large-scale distributed HPC and ML use cases.
+
+## How JobSet Works
+JobSet models a distributed batch workload as a group of Kubernetes Jobs. This allows a user to
+easily specify different pod templates for different distinct groups of pods (e.g. a leader,
+workers, parameter servers, etc.). 
+
+It uses the abstraction of a ReplicatedJob to manage child Jobs, where a ReplicatedJob is
+essentially a Job Template with some desired number of Job replicas specified. This provides a
+declarative way to easily create identical child-jobs to run on different islands of accelerators,
+without resorting to scripting or Helm charts to generate many versions of the same job but with
+different names.
+
+<figure>
+  <img src="jobset_diagram.svg">
+  <figcaption><h4>JobSet Architecture</h4></figcaption>
+</figure>
```

Comment:
```suggestion
{{< figure src="jobset_diagram.svg" alt="JobSet Architecture" class="diagram-large" clicktozoom="true" >}}
```

Reply:
can you please just apply the change?

Reply:
I'm happy for someone to pick this up, but I really don't want to be the single load-bearing part in getting blog articles published. Because of how much I value distributing the work, I don't plan to do this step myself. @ahg-g you are welcome to ask other contributors to help out here.

Reply:
Thanks @sftim, I was suggesting that Daniel applies the change :)